src/doubly_linked_list.js

```diff
@@ -0,0 +1,334 @@
+class DoubleNode {
+  constructor(data){
+    this.data = data
+    this.next = null
+    this.prev = null
+  }
+}
+
+
+export default class Doubly_linked_list {
+
+  constructor() {
+    this.size = 0
+    this.head = null
+    this.tail = null
+  }
+
+
+  insert(data) {
+
+    let doubleNode = new DoubleNode(data)
+
+    if(!this.head){
+      this.size++
+      return this.head = this.tail = doubleNode
+    }
+
+    let currentDoubleNode = this.head
+
+    while(currentDoubleNode.next){
+      currentDoubleNode = currentDoubleNode.next
+    }
+
+    currentDoubleNode.next = doubleNode
+    this.size++
+
+    let prevDoubleNode = this.head
+    let i = 0
+    while(prevDoubleNode && i < this.size-2){
```

Comment:
This looks like might be more concisely written as a for loop? The while loop above it is great, though